### DISCUSSION TITLE
How to create or renew traceId manually ?

### DISCUSSION THREAD

**Discussion by ldcsaa** [apache/skywalking#11189] [Q&A]
My scenario is: 
There is a JOB thread pool that wants to generate a traceId for each task execution .

**Comment by wu-sheng** [comment]
Once you create a new span without any tracing context, usually it is a case in the thread, unless you don't close the last tracing context(the last span) correctly. 
I can't follow your case, your asking is too general.

**Comment by ldcsaa** [comment]
For example:
There is a Job Handler inside the application that runs regularly. I hope it generates a traceId every time it executes, which is used to track the execution process of the job.

```
    @Scheduled(timeUnit = TimeUnit.SECONDS, fixedDelay = 3600)
    void jobHandler()
    {
    	/* generate a new traceId before jobHandler() execute */
    	// ...
    	
    	// get current traceId
    	String traceId = TraceContext.traceId();
    	
    	log.info("{} JOB start ...", traceId);
    	
    	// other codes ...
    	// traceId transfer to other service ...
    	
    	log.info("{} JOB end !", traceId);
    }
```
